crates/parser/src/grammar/expressions.rs

```diff
@@ -562,6 +562,25 @@ fn arg_list(p: &mut Parser) {
         if !expr_with_attrs(p) {
             break;
         }
+        if p.at(T![:]) {
+            // Eat a trailing `:` so the arg-list doesn't break if the user is typing a path
+
+            // test_err arg_recover_colon
+            // fn main() {
+            //     foo(y, x:, z);
+            // }
+            p.err_and_bump(&format!("expected {:?}", T![,]));
+        }
+        if p.at(IDENT) {
+            // Eat a ident so the arg-list doesn't break if the user is inserting a new param.
+
+            // test_err arg_recover_ident
+            // fn main() {
+            //     foo(y, x orig::fun(, z);
+            // }
+            p.error(&format!("expected {:?}", T![,]));
+            continue;
+        }
```

Comment:
I don't see the value in this, if the user types in a new argument I would expect them to add a `,` first

Reply:
I think it’s better to support both of the flavors. Personally I prefer typing the arg first then the dot.